common/reflect_resource_test.go

```diff
@@ -78,6 +78,33 @@ type testStruct struct {
 	Hidden2        string
 }
 
+type testRecursiveStruct struct {
+	Task  *testJobTask `json:"task,omitempty"`
+	Extra string       `json:"extra,omitempty"`
+}
+
+type testJobTask struct {
+	ForEachTask *testForEachTask `json:"for_each_task,omitempty"`
+	Extra       string           `json:"extra,omitempty"`
+}
+
+type testForEachTask struct {
+	Task  *testJobTask `json:"task,omitempty"`
+	Extra string       `json:"extra,omitempty"`
+}
```

Comment:
These structs above basically mimic the jobs `for_each_task` recursive structure that we are going to introduce